src/sqlfluff/core/rules/analysis/select_crawler.py

```diff
@@ -29,15 +40,40 @@ def gather(
         cls, segment: BaseSegment, dialect: Dialect
     ) -> Dict[Optional[str], List["SelectCrawler"]]:
         """Find top-level SELECTs and CTEs, return info."""
-        queries = defaultdict(list)
+        queries: DefaultDict[Optional[str], List[SelectCrawler]] = defaultdict(list)
+        cls._process_select_statements(segment, dialect, queries)
+        return dict(queries)
+
+    @classmethod
+    def _process_select_statements(
```

Comment:
New utility function, now used in two places (the new place handles nested CTEs)